### PR TITLE
chore(library/init/*): a few small improvements

### PR DESCRIPTION
**Pull request by urkud**
refs #160

* `inv_inv'` takes `a` as an explicit argument;
* `div_mul_left` for `division_ring`s;
* `prod.map` and `function.uncurry` don't use pattern matching;
* export `inhabited.default` instead of `def`ining it
* remove unneeded `fun.inhabited`.

# Pull Request Description

Ensure you have read the contribution guide before filling in a description of the
pull request, regardless of whether it is complete or a work in progress.
All Pull Requests should include test case(s) which demonstrates the intended
behavior of a feature, or a regression test demonstrating that the fix resolves
the issue.

### CODE REVIEW

**Comment by urkud** [comment]
Any chance to have it reviewed soon?

File: library/init/algebra/field.lean [hunk 1]
```diff
@@ -185,12 +193,26 @@ lemma one_div_div (a b : α) : 1 / (a / b) = b / a :=
 by rw [one_div_eq_inv, division_def, mul_inv',
        inv_inv', division_def]
 
+lemma div_helper {a : α} (b : α) (h : a ≠ 0) : (1 / (a * b)) * a = 1 / b :=
```
Comment (gebner) [inline library/init/algebra/field.lean]:
```suggestion
lemma div_mul_mul_eq_div {a : α} (b : α) (h : a ≠ 0) : (1 / (a * b)) * a = 1 / b :=
```

But I'm not confident of my ability to choose good names.

**Comment by cipher1024** [comment]
> So far I've always postponed actually merging the PRs until the release day. My reasoning is that it is easier to test changes to Lean if Lean master can build mathlib master.

If we switch to GitHub Actions, any chance we could get the build system to check mathlib for us?

**Comment by gebner** [comment]
> If we switch to GitHub Actions, any chance we could get the build system to check mathlib for us?

We could, but this check would typically fail.  Usually we make breaking changes in Lean, and then shortly before the release fix mathlib in a `lean3N` branch.

File: library/init/algebra/field.lean (hunk 1 above)
Comment (urkud) [reply library/init/algebra/field.lean]: I assumed that this is a name used by `norm_num` or something, so I didn't touch it.

Comment (gebner) [reply library/init/algebra/field.lean:196]: Oh wow. That's the name we already use. I guess it's ok then.

**Pull request merged**